File: leetcode/year_1/rand/rm_duplicates_sorted_linked_list.c
#include <stdio.h>

struct ListNode {
  int val;
  struct ListNode *next;
};

struct ListNode *deleteDuplicates(struct ListNode *head) {

  struct ListNode *curr_tail = head;
  struct ListNode *walk = head;

  while (walk) {
    if (walk->val != curr_tail->val) {
      curr_tail->next = walk;
      printf("added here\n");
      curr_tail = walk;
    }
    walk = walk->next;
  }

  curr_tail->next = NULL;

  return head;
}

File: leetcode/year_1/rand/rm_duplicates_sorted_linked_list.java
public class ListNode {
	int val;
	ListNode next;

	ListNode() {
	}

	ListNode(int val) {
		this.val = val;
	}

	ListNode(int val, ListNode next) {
		this.val = val;
		this.next = next;
	}
}

class Solution {
	public ListNode deleteDuplicates(ListNode head) {

		ListNode tail = head;

		ListNode walk = tail;

		while (walk != null) {
			if (walk.val != tail.val) {

				tail.next = walk;
				tail = walk;
			}
			walk = walk.next;
		}

		return tail;

	}
}
